pages/jobs/JobsFilter.jsx

```diff
@@ -0,0 +1,109 @@
+import {
+  Box,
+  Button,
+  Flex,
+  Heading,
+  Stack,
+  Text,
+  Wrap,
+  WrapItem,
+} from "@chakra-ui/react";
+import React, { useState } from "react";
+import { MdFilterList } from "react-icons/md";
+
+import { FilterTagButton } from "@/components/FilterTagButton";
+import { jobLocationsArray } from "@/utils/helpers/jobLocationsArray";
+import { jobTagsArray } from "@/utils/helpers/jobTagsArray";
+
+export const JobsFilter = ({
+  locationFiltersList,
+  tagFiltersList,
+  handleTagSelect,
+  handleLocationSelect,
+  handleClickRemoteOnly,
+}) => {
+  const [showMoreTags, setShowMoreTags] = useState(false);
+
+  const jobTagsByCategory = jobTagsArray.reduce((acc, curr) => {
+    if (!acc[curr.category]) {
+      acc[curr.category] = [];
+    }
+    acc[curr.category].push(curr);
+    return acc;
+  }, {});
+
+  return (
+    <Stack padding="8" borderWidth="1px" borderRadius="20px" spacing={4}>
+      <Box>
+        <Heading as="h5" marginBottom="4" size="md">
+          Narrow your search
+        </Heading>
+        <Button
+          leftIcon={<MdFilterList />}
+          onClick={handleClickRemoteOnly}
+          size="sm"
+        >
+          Show Remote Only
+        </Button>
+      </Box>
+      <Box>
+        <Text marginBottom={2}>Filter by location</Text>
+        <Wrap spacing={2}>
+          {jobLocationsArray.map((location) => (
+            <WrapItem key={location.value}>
+              <FilterTagButton
+                isSelected={locationFiltersList.some(
+                  (filter) => filter.value === location.value
+                )}
+                handleSelect={() => handleLocationSelect(location)}
+                name={location.label}
+              />
+            </WrapItem>
+          ))}
+        </Wrap>
+      </Box>
+      <Box>
+        <Text marginBottom={2}>Filter by tag</Text>
+        {/* Could create multiple wrap groups depending on how many tags want to be hidden */}
+        <Wrap spacing={2}>
+          {showMoreTags
+            ? Object.entries(jobTagsByCategory).map(([category, tags]) => (
```

Comment:
I like the way you handled this and it got me thinking, let's maybe show the first 6 filters you have there (front end, back end, full stack, and full time, part time, contract) and hide the rest - maybe pop them in alphabetical order, too? 

Another thing I was thinking of that could be neat that we can split out into it's own thing is when you click the "Extra" filters it'll only show filters that are actively applied to jobs. An example would be Express, it's in the list, but there are no Express jobs so not even showing that would be kinda neat. What do you think?

Reply:
1. This has been added. Thank you for the suggestion.
2. Yea, lets record this for a follow on to this